controllers/autoscaling.go

```diff
@@ -18,30 +19,58 @@ func (r *HorizontalRunnerAutoscalerReconciler) determineDesiredReplicas(rd v1alp
 
 	var repos [][]string
 
-	repoID := rd.Spec.Template.Spec.Repository
-	if repoID == "" {
-		orgName := rd.Spec.Template.Spec.Organization
-		if orgName == "" {
-			return nil, fmt.Errorf("asserting runner deployment spec to detect bug: spec.template.organization should not be empty on this code path")
-		}
+	orgName := rd.Spec.Template.Spec.Organization
+	if orgName == "" {
+		return nil, fmt.Errorf("asserting runner deployment spec to detect bug: spec.template.organization should not be empty on this code path")
+	}
+
+	metrics := hra.Spec.Metrics
+	if len(metrics) == 0 {
+		return nil, fmt.Errorf("validating autoscaling metrics: one or more metrics is required")
+	} else if tpe := metrics[0].Type; tpe != v1alpha1.AutoscalingMetricTypeTotalNumberOfQueuedAndInProgressWorkflowRuns {
+		return nil, fmt.Errorf("validting autoscaling metrics: unsupported metric type %q: only supported value is %s", tpe, v1alpha1.AutoscalingMetricTypeTotalNumberOfQueuedAndInProgressWorkflowRuns)
+	}
 
-		metrics := hra.Spec.Metrics
+	if len(metrics[0].RepositoryNames) == 0 {
```

Comment:
somethings to consider:
1. https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#list-selected-repositories-enabled-for-github-actions-in-an-organization is available if not using GHE. And most likely will become available in actions GA sometime in Q4.

2. This change should be feature flagged

Reply:
Thanks for your input! 

RE #1: Yep, good point. We obviously need to check repos for which Actions is enabled. Will look into this tomorrow / next week.
RE #2: I agree, but I don't have the permissions to add labels to PRs.